src/main/scala/firrtl/passes/memlib/MemIR.scala

```diff
@@ -0,0 +1,43 @@
+// See LICENSE for license details.
+
+package firrtl.passes
+package memlib
+
+import firrtl._
+import firrtl.ir._
+import Utils.indent
+
+case class DefAnnotatedMemory(
+    info: Info,
+    name: String,
+    dataType: Type,
+    depth: Int,
+    writeLatency: Int,
+    readLatency: Int,
+    readers: Seq[String],
+    writers: Seq[String],
+    readwriters: Seq[String],
+    readUnderWrite: Option[String],
+    maskGran: Option[BigInt],
+    memRef: Option[String]
+    //pins: Seq[Pin],
+    ) extends Statement with IsDeclaration {
+  def serialize: String =
+    s"mem $name :" + info.serialize +
+    indent(
+      (Seq("\ndata-type => " + dataType.serialize,
+          "depth => " + depth,
+          "read-latency => " + readLatency,
+          "write-latency => " + writeLatency) ++
+          (readers map ("reader => " + _)) ++
+          (writers map ("writer => " + _)) ++
+          (readwriters map ("readwriter => " + _)) ++
+       Seq("read-under-write => undefined")) mkString "\n")
```

Comment:
When read-under-write annotations actually get passed through from Chisel, it'd be nice if you only had to change in 1 place...